docs/sources/http_api/data_source.md

```diff
@@ -712,3 +714,93 @@ Content-Type: application/json
   }
 }
 ```
+
+## Query a data source by ID
+
+Queries a data source having backend implementation.
+
+`POST /api/ds/query`
+
+> **Note:** Most of Grafana's builtin data sources have backend implementation.
+
+**Example request for the Test data source**:
+
+```http
+POST /api/ds/query HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+
+{
+   "queries":[
+      {
+         "refId":"A",
+         "scenarioId":"csv_metric_values",
+         "datasource":{
+            "uid":"PD8C576611E62080A",
+            "type":"testdata"
+         },
+         "stringInput":"1,20,90,30,5,0",
+         "datasourceId":6,
+         "maxDataPoints":1848,
+         "intervalMs":200,
+      }
+   ],
+   "from":"now-5m",
+   "to":"now"
+}
+```
+
+> **Note:** The `from`, `to`, and `queries` properties are required.
+
+JSON Body schema:
+
+- **queries** – Specifies one or more queries. Must contain at least 1.
+- **queries.refId** – Specifies an identifier of the query. Is optional and default to "A".
+- **queries.datasource.uid** – Specifies the UID of data source to be queried. Each `query` in the request must have an unique `datasource`.
+- **queries.datasource.type** – Specifies the data source plugin identifier to be queried.
+- **queries.maxDataPoints** - Species maximum amount of data points that dashboard panel can render. Is optional and defaults to 100.
+- **queries.intervalMs** - Specifies the time interval in milliseconds of time series. Is optional and defaults to 1000.
+- **from/to** – Should be either absolute in epoch timestamps in milliseconds or relative using Grafana time units. For example, `now-5m`.
+
+  In addition, each data source has its own specific properties that should be added in a request.
```

Comment:
Wonder if we can describe this in a way that's still helpful for users compared with how we've documented /api/tsdb/query that included some other examples, e.g. mysql. Maybe we can refer to something a long the lines "Please use the Developer Tools of your browser of choice and inspect the HTTP requests being made to `/api/ds/query` to understand how to query a certain datasource in question."?

Not sure if these spaces was intentional?

```suggestion
In addition, each data source has its own specific properties that should be added in a request.
```